### PR TITLE
Added validation of fixed/free handicap input

### PR DESCRIPTION
**Pull request by sethtroisi**
"place_free_handicap" without specifying a number of stones results in leelaz adding an arbitrary number of handicap stones.

### CODE REVIEW

File: src/GTP.cpp [hunk 1]
```diff
@@ -544,10 +544,13 @@ bool GTP::execute(GameState & game, std::string xinput) {
         cmdstream >> tmp;   // eat place_free_handicap
         cmdstream >> stones;
 
-        game.place_free_handicap(stones);
-
-        std::string stonestring = game.board.get_stone_list();
-        gtp_printf(id, "%s", stonestring.c_str());
+        if (!cmdstream.fail()) {
+            game.place_free_handicap(stones);
+            std::string stonestring = game.board.get_stone_list();
```
Comment (gcp) [inline src/GTP.cpp]: nit: prefer auto

File: src/GTP.cpp [hunk 2]
```diff
@@ -544,10 +544,13 @@ bool GTP::execute(GameState & game, std::string xinput) {
...
+            gtp_printf(id, "%s", stonestring.c_str());
+        } else {
+            gtp_fail_printf(id, "Not a valid number of handicap stones");
```
Comment (gcp) [inline src/GTP.cpp]: I think this is the wrong error message in this case. "Not a valid list of vertices" or something similar?

Comment (sethtroisi) [reply src/GTP.cpp]: If you look five lines above you'll see it reads an integer "stones" from cmdstream, so I think this is the right error message.

Also see https://www.gnu.org/software/gnugo/gnugo_19.html
"Fails:     invalid number of stones"

**Comment by sethtroisi** [comment]
updated to using auto

**Pull request merged**